react/features/always-on-top/AlwaysOnTop.js

```diff
@@ -0,0 +1,297 @@
+     * @returns {void}
+     */
+    _audioAvailabilityListener({ available }) {
+        this.setState({ audioAvailable: available });
+    }
+
+    /**
+     * Handles audio muted api events.
+     *
+     * @param {{ muted: boolean }} status - The new muted status.
+     * @returns {void}
+     */
+    _audioMutedListener({ muted }) {
+        this.setState({ audioMuted: muted });
+    }
+
+    /**
+     * Hides the toolbar after a timeout.
+     *
+     * @returns {void}
+     */
+    _hideToolbarAfterTimeout() {
+        setTimeout(() => {
+            if (this._hovered) {
+                this._hideToolbarAfterTimeout();
+
+                return;
+            }
+            this.setState({ visible: false });
+        }, TOOLBAR_TIMEOUT);
+    }
+
+    /**
+     * Handles mouse move events.
+     *
+     * @returns {void}
+     */
+    _mouseMove() {
+        if (!this.state.visible) {
+            this.setState({ visible: true });
+        }
+    }
+
+    /**
+     * Toolbar mouse over handler.
+     *
+     * @returns {void}
+     */
+    _onMouseOver() {
+        this._hovered = true;
+    }
+
+    /**
+     * Toolbar mouse out handler.
+     *
+     * @returns {void}
+     */
+    _onMouseOut() {
+        this._hovered = false;
+    }
+
+    /**
+     * Handles audio available api events.
+     *
+     * @param {{ available: boolean }} status - The new available status.
+     * @returns {void}
+     */
+    _videoAvailabilityListener({ available }) {
+        this.setState({ videoAvailable: available });
+    }
+
+    /**
+     * Handles video muted api events.
+     *
+     * @param {{ muted: boolean }} status - The new muted status.
+     * @returns {void}
+     */
+    _videoMutedListener({ muted }) {
+        this.setState({ videoMuted: muted });
+    }
+
+    /**
+     * Sets mouse move listener and initial toolbar timeout.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        api.on('audioMuteStatusChanged', this._audioMutedListener);
+        api.on('videoMuteStatusChanged', this._videoMutedListener);
+        api.on('audioAvailabilityChanged', this._audioAvailabilityListener);
+        api.on('videoAvailabilityChanged', this._videoAvailabilityListener);
+
+        Promise.all([
+            api.isAudioMuted(),
+            api.isVideoMuted(),
+            api.isAudioAvailable(),
+            api.isVideoAvailable()
+        ])
+        .then(([
+            audioMuted = false,
+            videoMuted = false,
+            audioAvailable = false,
+            videoAvailable = false
+        ]) =>
+            this.setState({
+                audioMuted,
+                videoMuted,
+                audioAvailable,
+                videoAvailable
+            })
+        )
+        .catch(console.error);
+
+        window.addEventListener('mousemove', this._mouseMove);
+
+        this._hideToolbarAfterTimeout();
+    }
+
+    /**
+     * Removes the mouse move listener.
```

Comment:
I think this comment needs updating because more than the mouse move listener is removed.